src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -234,6 +231,37 @@ public ResponseEntity<StudentExam> updateWorkingTime(@PathVariable Long courseId
         return ResponseEntity.ok(savedStudentExam);
     }
 
+    /**
+     * POST /courses/{courseId}/exams/{examId}/student-exams/{studentExamId}/attendance-check : Throw attandance Check Event in the student exam
+     *
+     * @param courseId     the course to which the student exams belong to
+     * @param examId       the exam to which the student exams belong to
+     * @param studentLogin the id of the student exam to find
+     * @return the ResponseEntity with status 200 (OK) and with the updated student exam as body
+     */
+    @PostMapping("/courses/{courseId}/exams/{examId}/students/{studentLogin:" + Constants.LOGIN_REGEX + "}/attendance-check")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<ExamAttendanceCheckEventDTO> attendanceCheck(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable String studentLogin,
+            @RequestBody(required = false) String message) {
+        log.debug("REST request for attendance-check for student : {}", studentLogin);
+
+        var student = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin)
+                .orElseThrow(() -> new EntityNotFoundException("User with login: \"" + studentLogin + "\" does not exist"));
```

Comment:
We often have "ElseThrow" variants of the database retrievals. Can you also add one for this query and replace the other calls to this method as well?